### PR TITLE
Feature/The Giving Blocks

### PR DESCRIPTION
**Pull request by CarlosQ96**
Issue: #160 
Cronjob to synchronize all the projects of the givingBlocks.

Known Issues:
Axios Hangs on subsequent requests even with delays.
Will run the process 3 times to ensure everything is created correctly.

Docs: https://the-giving-block.gitbook.io/public-api-documentation/

### CODE REVIEW

**Comment by mohammadranjbarz** [comment]
Thanks @CarlosQ96 Its seems great, but I had some change requests that I made in https://github.com/Giveth/impact-graph/pull/236, please review and merge that

File: services/the-giving-blocks/api.ts [hunk 1]
```diff
@@ -0,0 +1,130 @@
...
+  } catch (e) {
+    console.log('giving block service fetchGivingBlockProjects() err', e);
+    throw e;
+  }
+};
+
+export const fetchOrganizationById = async (
+  accessToken: string,
+  organizationId: number,
+) => {
```
Comment (mohammadranjbarz) [inline services/the-giving-blocks/api.ts]: Could you please specify output type for this function?

File: services/the-giving-blocks/syncProjectsCronJob.ts [hunk 2]
```diff
@@ -0,0 +1,183 @@
...
+};
+
+// The GivingBlocks itself will be the category
+// To avoid extra logic bringing external ones.
+type GivingBlocksCategory = {
+  id: number;
+  name: string;
+};
+
+const findOrCreateGivingBlocksCategory = async () => {
```
Comment (mohammadranjbarz) [inline services/the-giving-blocks/syncProjectsCronJob.ts]: @CarlosQ96  please specify output type for this function

File: services/the-giving-blocks/syncProjectsCronJob.ts [hunk 3]
```diff
@@ -0,0 +1,183 @@
...
+      await ProjectUpdate.save(update);
+    }
+  } catch (e) {
+    // Log Error but keep going with the rest of the projects
+    console.log(e);
+  }
+};
+
+// Current Formula: This will be changed in the future
+const getQualityScore = (description, hasImageUpload) => {
```
Comment (mohammadranjbarz) [inline services/the-giving-blocks/syncProjectsCronJob.ts]: @CarlosQ96  please specify output type for this function

File: services/the-giving-blocks/syncProjectsCronJob.ts [hunk 4]
```diff
@@ -0,0 +1,183 @@
...
+  let qualityScore = 40;
+
+  // Some projects have no description
+  if (Number(description?.length) > 100) qualityScore = qualityScore + 10;
+  if (hasImageUpload) qualityScore = qualityScore + 30;
+
+  return qualityScore;
+};
+
+const getAppropriateSlug = async (slugBase: string) => {
```
Comment (mohammadranjbarz) [inline services/the-giving-blocks/syncProjectsCronJob.ts]: @CarlosQ96  please specify output type for this function
PS: We used that before, that time we didn't write type, but it's good to write types from now on

**Review by mohammadranjbarz — CHANGES_REQUESTED** [review]
Thanks @CarlosQ96 
Please applly changes requested, I would test the functionality later

**Comment by CarlosQ96** [comment]
@mohammadranjbarz made the changes. Ready to be merged!

**Review by mohammadranjbarz — APPROVED** [review]
Amazing work @CarlosQ96 
I have tested it and the functionality was ok

**Pull request merged**